[DNM] Upmerge 04 11 2019

de-nordic opened the pull request.

carlescufi committed:
[nrf noup] Re-apply downstream changes to mcumgr

As mbedtls is no longer part of the Zephyhr's codebase, and we had it
modified downstream, we need to re-apply those changes in this
repository.

This is the mbedTLS portion of
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/110
after mbedTLS was split out from the zephyr ext/ folder into its own
repository.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
(cherry picked from commit 900c7eea92e45adba1067e044ec705d4541c84a5)
Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

vanwinkeljan committed:
native: debug: Set speed optimization if ASAN is used

Set speed optimization, instead of size, in case address sanitizer is
enabled for native_posix. This is need as the i386 assembly code used
in mbetls does not compile if address sanitizer is used together with
size optimization.

Signed-off-by: Jan Van Winkel <jan.van_winkel@dxplore.eu>

SebastianBoe committed:
cmake: Misc. cleanup of build scripts

Misc. cleanup of build scripts. Most importantly, mbedTLS is no longer
linked with zephyr_interface. It should not be as there is no reason
to do so. This aligns with
https://github.com/zephyrproject-rtos/zephyr/pull/19346

Also, glob the source files in 'library' instead of listing them to
make it clear that all files are used.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/NordicPlayground/fw-nrfconnect-mbedtls?pullRequest=5) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/NordicPlayground/fw-nrfconnect-mbedtls?pullRequest=5) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: SebastianBoe<br/>:white_check_mark: carlescufi<br/>:white_check_mark: de-nordic<br/>:x: vanwinkeljan<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/NordicPlayground/fw-nrfconnect-mbedtls?pullRequest=5) it.</sub>

mbolivar reviewed (changes requested):
@de-nordic
> [nrf upmerge] Merge commit from zephyrproject to NRF 54fb2eb

Should be `[nrf mergeup]`, but looks good otherwise

de-nordic committed:
Revert "[nrf noup] Re-apply downstream changes to mcumgr"

This reverts commit 900c7eea92e45adba1067e044ec705d4541c84a5.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
[nrf mergeup] Merge commits from zephyrproject to NRF

Merge from upstream zephyrproject-rtos/mbedtls:4f1e8f5a7 to
fw-nrfconnect-mbedtls:aa991493

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic commented:
@SebastianBoe hey, can you look. I have rebased the marge to include your changes for multi-image support.
